### DISCUSSION TITLE
LPIPS gives negative value

### DISCUSSION THREAD

**Discussion by mshooter** [Lightning-AI/torchmetrics#2310] [CompVision]
Hi everyone, 

my pseudo code is 
```
loss = LearnedPerceptualImagePatchSimilarity(net_type="vgg", normalize=True) 
y_hat = model(x) 
loss(x, y_hat) 
```
both x and y_hat have values between [0,1]. 
But I don't understand why my loss is negative..

**Comment by F-Barto** [comment]
Same but with input values between [-1,1].
I am using it as a training loss.

![image](https://github.com/Lightning-AI/torchmetrics/assets/905829/9c65e405-7616-4c6e-9ac8-b8f95275e5fb)

**Comment by F-Barto** [comment]
Just opened an issue: #2315

**Comment by Borda** [comment]
@mshooter @F-Barto — this is a known issue and there are two common causes:

**Cause 1: Input range mismatch.** This is the most common one.
- `normalize=False` (default) expects inputs in **[-1, 1]**
- `normalize=True` expects inputs in **[0, 1]** and internally rescales to [-1, 1]

If your inputs are in [0, 1] but `normalize=False`, the network sees only half the expected range and can produce negative scores.
```python
from torchmetrics.image.lpip import LearnedPerceptualImagePatchSimilarity

# If your images are [0, 1]:
lpips = LearnedPerceptualImagePatchSimilarity(net_type="vgg", normalize=True)

# If your images are [-1, 1]:
lpips = LearnedPerceptualImagePatchSimilarity(net_type="vgg", normalize=False)
```

**Cause 2: Using LPIPS as a training loss.** When backpropagating through the LPIPS network, the learned linear layers can yield negative values. The LPIPS paper states scores should be non-negative for detached evaluation, but during training gradient flow can push intermediate values negative.

**Workaround if you need LPIPS as a loss:**
```python
loss = lpips(pred, target)
loss = torch.clamp(loss, min=0.0)  # safety clamp
```

**Verify your setup:**
```python
import torch
img1 = torch.rand(4, 3, 64, 64)  # [0, 1] range
img2 = torch.rand(4, 3, 64, 64)

lpips = LearnedPerceptualImagePatchSimilarity(net_type="vgg", normalize=True)
score = lpips(img1, img2)
print(score)  # should be positive, typically 0.0–1.0
```

Docs: [LPIPS](https://lightning.ai/docs/torchmetrics/stable/image/learned_perceptual_image_patch_similarity.html)
